src/runtime/hexagon/hexagon_hvx.cc

```diff
@@ -31,25 +31,30 @@ namespace tvm {
 namespace runtime {
 namespace hexagon {
 
-HexagonHvx::HexagonHvx() {
-  // Reserve HVX.
-  int res = qurt_hvx_reserve(QURT_HVX_RESERVE_ALL_AVAILABLE);
-  CHECK((res != QURT_HVX_RESERVE_NOT_SUPPORTED) && (res != QURT_HVX_RESERVE_NOT_SUCCESSFUL))
-      << "error reserving HVX: " << res;
+HexagonHvx::HexagonHvx() { Acquire(); }
 
-  // Lock HVX.
+HexagonHvx::~HexagonHvx() { Release(); }
+
+void HexagonHvx::Acquire() {
+  reserved_count_ = qurt_hvx_reserve(QURT_HVX_RESERVE_ALL);
```

Comment:
This explicitly reserves all of the HVX units.  Depending on schedules, we may want to request fewer HVX units so a future change could include requesting only the number of HVX instances needed.  Thoughts on this approach?

Reply:
I think we'll always want to own the set of available HVX coprocessors and so this is good.

Reply:
Oh wait, does every thread main call `qurt_hvx_reserve(QURT_HVX_RESERVE_ALL);` because each owns an HexagonHvx?

Reply:
No, the main thread constructs a HexagonHvx object, which calls qurt_hvx_reserve.  Each thread assigned to an HVX_x instance calls Lock();